### ISSUE TITLE
Lock screen feature like on Garmin devices

### ISSUE THREAD

**Issue by albertgardu** [osmandapp/OsmAnd#20742]
### Describe the idea

**Problem 1:** while riding with my motorbike and using OsmAnd, if it rains, the raindrops "touch" the screen randomly making it very hard to follow the route because they might exit the app, or activate some feature.

**Problem 2:** while riding with my motorbike offroad and using OsmAnd, sometimes, the screen gets covered by dust and the easiest way to clean it on the go is by doing it with your hands (normally with gloves). However, when swiping the hand over the screen, usually you "touch" the screen and exit the app, or activate some feature.

**Solution:** add a new function to OsmAnd that locks the screen (without turning it off). All Garmin GPS devices have this function, or roadbook apps like PI Roadbook.

### Expected behaviour

A floating "lock/unlock" icon on the OsmAnd screen could work. The only issue I see is that a raindrop could also unlock it at some point.

Even better if the user could assign one of the phone/tablet physical buttons to do the locking/unlocking action.

![image](https://github.com/user-attachments/assets/48a212fe-9e5f-4d91-b289-7febe369e49a)

### Alternatives you've considered

Some riders we use additional/external bluetooth devices on our handlebar to use basic functions of OsmAnd (like center the map, or zoom in/out). It would be great to be able to assign the lock screen feature to one of the button of the external device.

### Context

Example of an external bluetooth device to use basic OsmAnd functions while riding: https://www.arnrally.com/en/producto/arn-remote-pro/

**Comment by mariush444** [comment]
https://github.com/osmandapp/OsmAnd/issues/6770

**Comment by vshcherb** [comment]
We can add quick action to lock screen, so we can add custom button

**Comment by albertgardu** [comment]
@vshcherb that would be awesome!

**Comment by dmpr0** [comment]
## ToDo

<table style="width:100%">
  <tr>
    <td><img src="https://github.com/user-attachments/assets/9edd3e67-8f9d-477c-bd90-3e02358faede"  width="150" height="auto"/> </td>
  </tr>
</table>   

- [ ] Add new quick action into **Interface** group
- [ ] Icon – `ic_action_touch_screen_lock`
- [ ] Name – `Turn On / Off Touch screen lock`
- [ ] Description – `Enables or disables touch screen functionality to prevent accidental taps`

**Comment by ossilator** [comment]
to avoid the "recursive" problem that drops could trigger the unlock button, make it a slide switch, or a button that needs a subsequent swipe.

**Comment by cricri-du-lauragais** [comment]
i use android app "touch lock", work fine for my galaxy a33

**Comment by dmpr0** [comment]
# Review

- [ ] Show toast messages:
- - [ ] After screen is locked – `Touch screen is locked. Long press on any button to unlock.`
- - [ ] After screen in unlocked – `Touch screen is unlocked.`
- [ ] If screen was locked by external button, include button name in description `Touch screen is locked. Press "Volume down" button to unlock, or long press on any on screen buttons.`

**Comment by shurasmiling** [comment]
+ https://github.com/osmandapp/OsmAnd/issues/20711

**Comment by shurasmiling** [comment]
- https://osmand.net/docs/user/widgets/quick-action/#interface
- https://osmand.net/docs/user/widgets/quick-action/#touch-screen-lock
